core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -690,6 +713,40 @@ func (vdb *VersionedDB) Close() {
 	// no need to close db since a shared couch instance is used
 }
 
+// updateAndSaveNamespaceDBsInfo updates and saves vdb.namespaceDBsInfo if the dbName is not in namespaceDBsInfo.
+// Otherwise, it does nothing.
+func (vdb *VersionedDB) updateAndSaveNamespaceDBsInfo(ns, dbName string) error {
+	if _, ok := vdb.namespaceDBsInfo[ns]; ok {
+		return nil
+	}
+
+	logger.Debugf("[%s] add namespaceDBInfo for namespace %s", vdb.chainName, ns)
+	vdb.namespaceDBsInfo[ns] = &namespaceDBInfo{
+		Namespace: ns,
+		DBName:    dbName,
+	}
+	dbsInfo := make([]*namespaceDBInfo, 0, len(vdb.namespaceDBsInfo))
+	for _, dbInfo := range vdb.namespaceDBsInfo {
+		dbsInfo = append(dbsInfo, dbInfo)
+	}
+	couchDoc, err := encodeChannelMetadata(dbsInfo)
+	if err != nil {
+		return err
+	}
+	if _, err := vdb.metadataDB.saveDoc(channelMetadataDocID, "", couchDoc); err != nil {
+		return err
+	}
+	dbResponse, err := vdb.metadataDB.ensureFullCommit()
+	if err != nil {
+		return err
+	}
+	if !dbResponse.Ok {
+		logger.Errorf("failed to perform full commit while saving couchdb metadata")
+		return errors.New("failed to perform full commit while saving couchdb metadata")
+	}
```

Comment:
1. The `ensureFullCommit()` checks for `dbResponse.Ok` and return an error if it is false. Hence, we don't need to repeat the same check here. 
2. In a separate PR, we can refactor `ensureFullCommit()` as it returns the dbResponse unnecessarily which leads to such code duplication.